is_3d: false
particle_filter:
  num_particles: 16
  resample_threshold: 0.50   #From 0 to 1
  covariance_coefficient: 0.0
  still_threshold: 0.01
  variance:
    x: 0.15 # X veloctiy in robot's frame
    y: 0.0000001 # Y velocty in robot's frame (AKA should be zero, but idk)
    yaw: 0.08
profile: false
filtering:
  voxel_grid_size: 0.1
  voxel_filter_enabled: true
scanmatcher:
  enable: true
  use_guess: true
  point_threshold: 0.5
  variance:
    x: 0.01
    y: 0.01
    yaw: 0.01
  icp:
    iterations: 50
    RANSAC_outlier_rejection_threshold: 0.001
    max_correspondence_distance: 0.1
    transformation_epsilon: 0.00001
  optimizer:
    angular_delta_start: 0.5
    linear_delta_start: 0.25
    max_refinements: 16
    kernel_size: 4
    gaussian_sigma: 0.1
icp:
  match_history_length: 1
visualization:
  hue: #0 to 360
    start: 257.0
    end: 12.0
  saturation: # 0 to 100
    start: 75
    end: 100
  lightness:
    start: 70
    end: 60
transform_max_wait_time: 1 # Seconds I think?
update_time_threshold: 0.2
start_x: 100
start_y: 100
start_z: 0

particle_filter_buffer_size: 64
pose_buffer_size: 64

debug: true

lidar_topic: "/scan/pointcloud"
fused_topic: "/odometry/filtered"
odometry_frame: "/odom"
base_frame: "/base_link"
lidar_frame: "/lidar"
parent_frame: "/odom"
child_frame: "/particle_filter"
